specification/notificationhubs/Notificationhubs.Management/NamespaceResource.tsp

```diff
@@ -0,0 +1,152 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "./models.tsp";
+
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.NotificationHubs;
+/**
+ * Notification Hubs Namespace Resource.
+ */
+@Azure.ResourceManager.Private.armResourceInternal(NamespaceProperties)
+@Http.Private.includeInapplicableMetadataInPayload(false)
+model NamespaceResource extends Foundations.TrackedResource {
+  ...ResourceNameParameter<
+    Resource = NamespaceResource,
+    KeyName = "namespaceName",
+    SegmentName = "namespaces",
+    NamePattern = "^[a-zA-Z][a-zA-Z0-9-]*$"
+  >;
+
+  /**
+   * The Sku description for a namespace
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "For backward compatibility"
+  sku: Sku;
+
+  @doc("The resource-specific properties for this resource.")
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  @Azure.ResourceManager.Private.armResourcePropertiesOptionality(true)
+  properties?: NamespaceProperties;
+}
+
+@armResourceOperations
+interface NamespaceResources {
+  /**
+   * Returns the given namespace.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-openapi" "non-standard operations"
+  @operationId("Namespaces_Get")
```

Comment:
Should we rename the interface to `Namespaces` instead?

Reply:
I don't think I'll change anything until we have a mature design and mitigation plan on the deprecating operation id.